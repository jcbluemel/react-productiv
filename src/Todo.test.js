import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

const TEST_TODO = {
  id: "1",
  title: "Test",
  description: "test",
  priority: 1
}

it("renders without crashing", function() {
  render(<Todo todo={TEST_TODO} />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<Todo todo={TEST_TODO} />);
  expect(asFragment()).toMatchSnapshot();
});

it("contains todo data", function() {
  const { container } = render(<Todo todo={TEST_TODO} />);
  expect(container).toContainHTML('class="Todo" id="1"');
  expect(container).toContainHTML('<b>Test</b>');
  expect(container).toContainHTML('<small>Priority: 1</small>');
  expect(container).toContainHTML('<small>test</small>');
});