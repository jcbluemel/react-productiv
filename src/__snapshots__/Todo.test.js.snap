// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
        Test
      </b>
       
      <small>
        Priority: 1
      </small>
    </div>
    <div>
      <small>
        test
      </small>
    </div>
  </div>
</DocumentFragment>
`;
